src/terminal/Screen_test.cpp

```diff
@@ -2911,11 +2911,34 @@ TEST_CASE("OSC.4")
 
     SECTION("query")
     {
-        screen.write("\033]4;1;?");
+        screen.write("\033]4;7;?\033\\");
         INFO(term.replyData);
+        REQUIRE(e(term.replyData) == e("\033]4;7;rgb:c0c0/c0c0/c0c0\033\\"));
     }
 
-    SECTION("set") { screen.write("\033]4;1;rgb:ab/cd/ef"); }
+    SECTION("set color via format rgb:RR/GG/BB")
+    {
+        screen.write("\033]4;7;rgb:ab/cd/ef\033\\");
+        screen.write("\033]4;7;?\033\\");
+        INFO(term.replyData);
+        REQUIRE(e(term.replyData) == e("\033]4;7;rgb:abab/cdcd/efef\033\\"));
+    }
+
+    SECTION("set color via format #RRGGBB")
+    {
+        screen.write("\033]4;7;#abcdef\033\\");
+        screen.write("\033]4;7;?\033\\");
+        INFO(term.replyData);
+        REQUIRE(e(term.replyData) == e("\033]4;7;rgb:abab/cdcd/efef\033\\"));
+    }
+
+    SECTION("set color via format #RGB")
+    {
+        screen.write("\033]4;7;#abc\033\\");
+        screen.write("\033]4;7;?\033\\");
+        INFO(term.replyData);
+        REQUIRE(e(term.replyData) == e("\033]4;7;rgb:aaaa/bbbb/cccc\033\\"));
```

Comment:
`urxvt` (rxvt-unicode) replies with `rgb:a000/b000/c000` if the color was set as `#abc`

Reply:
`xterm` actually replies with `rgb:a0a0/b0b0/c0c0` and Kitty replies with `rgb:aaaa/bbbb/cccc`. 
But you are right, `XParseColor` man page states that the `#`-format does not scale the value but instead just uses the parsed components as most significant bits, leaving out the rest with zeros.
Now I am feeling a little bit in the middle. I am inclined to support it the way xterm does, because internally Contour's RGB(A) is dealing with 8-bit component values only anyways.

I've changed the code to match xterm.